packages/server/src/models/UserItemModel.ts

```diff
@@ -116,47 +123,53 @@ export default class UserItemModel extends BaseModel<UserItem> {
 		await this.deleteBy({ byShareId: shareId, byUserId: userId });
 	}
 
-	public async add(userId: Uuid, itemId: Uuid, options: SaveOptions = {}): Promise<void> {
+	public async add(userId: Uuid, itemId: Uuid, options: UserItemSaveOptions = {}): Promise<void> {
 		const item = await this.models().item().load(itemId, { fields: ['id', 'name'] });
 		if (!item) {
 			throw new ErrorNotFound(`No such item: ${itemId}`);
 		}
 		await this.addMulti(userId, [item], options);
 	}
 
-	public async addMulti(userId: Uuid, itemsQuery: Knex.QueryBuilder | Item[], options: SaveOptions = {}): Promise<void> {
+	public async addMulti(userId: Uuid, itemsQuery: Knex.QueryBuilder | Item[], options: UserItemSaveOptions = {}): Promise<void> {
 		const perfTimer = new PerformanceTimer(logger, 'addMulti');
 
 		perfTimer.push('Main');
 
 		const items: Item[] = Array.isArray(itemsQuery) ? itemsQuery : await itemsQuery.whereNotIn('id', this.db('user_items').select('item_id').where('user_id', '=', userId));
 		if (!items.length) return;
 
-		await this.withTransaction(async () => {
-			perfTimer.push(`Processing ${items.length} items`);
-
-			for (const item of items) {
-				if (!('name' in item) || !('id' in item)) throw new Error('item.id and item.name must be set');
+		perfTimer.push(`Processing ${items.length} items`);
 
-				await super.save({
-					user_id: userId,
-					item_id: item.id,
-				}, options);
+		for (const item of items) {
+			if (!('name' in item) || !('id' in item)) throw new Error('item.id and item.name must be set');
 
-				if (this.models().item().shouldRecordChange(item.name)) {
-					await this.models().change().save({
-						item_type: ItemType.UserItem,
-						item_id: item.id,
-						item_name: item.name,
-						type: ChangeType.Create,
-						previous_item: '',
+			await this.withTransaction(async () => {
```

Comment:
This `withTransaction` has been moved within the `for` loop to avoid the following error observed while running tests in CI: `current transaction is aborted, commands ignored until end of transaction block`.